listenbrainz/webserver/static/js/src/CBReviewModal.tsx

```diff
@@ -0,0 +1,703 @@
+import * as React from "react";
+import { get as _get } from "lodash";
+
+import { Rating } from "react-simple-star-rating";
+
+import * as iso from "@cospired/i18n-iso-languages";
+import * as eng from "@cospired/i18n-iso-languages/langs/en.json";
+import * as _ from "lodash";
+import { faInfoCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import GlobalAppContext from "./GlobalAppContext";
+
+import { countWords } from "./utils";
+import Loader from "./components/Loader";
+
+export type CBReviewModalProps = {
+  listen: Listen;
+  isCurrentUser: Boolean;
+  newAlert: (
+    alertType: AlertType,
+    title: string,
+    message: string | JSX.Element
+  ) => void;
+};
+
+export interface CBReviewModalState {
+  entityToReview: ReviewableEntity | null;
+  loading: boolean;
+  reviewValidateAlert: string | null;
+
+  releaseGroupEntity: ReviewableEntity | null;
+  artistEntity: ReviewableEntity | null;
+  recordingEntity: ReviewableEntity | null;
+
+  textContent: string;
+  rating: number;
+  language: string;
+  acceptLicense: boolean;
+
+  success: boolean;
+  reviewMBID?: string;
+}
+
+iso.registerLocale(eng); // library requires language of the language list to be initiated
+
+export default class CBReviewModal extends React.Component<
+  CBReviewModalProps,
+  CBReviewModalState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  readonly minTextLength = 25;
+  readonly maxTextLength = 100000;
+
+  private CBBaseUrl = "https://critiquebrainz.org"; // only used for href
+  private MBBaseUrl = "https://metabrainz.org"; // only used for href
+  private allLanguagesKeyValue = Object.entries(iso.getNames("en")); // gets all languages
+
+  private CBInfoButton = (
+    <button
+      type="button"
+      className="btn-transparent capitalize-bold"
+      data-toggle="popover"
+      title="What is CritiqueBrainz?"
+      data-placement="bottom"
+      data-html="true"
+      data-content={`CritiqueBrainz is a <a href='${this.MBBaseUrl}/projects'>
+      MetaBrainz project</a> aimed at providing an open platform for music critics
+      and hosting Creative Commons licensed music reviews. </br></br>
+      Your reviews will be independently visible on CritiqueBrainz and appear publicly 
+      on your CritiqueBrainz profile. To view or delete your reviews, visit your 
+      <a href='${this.CBBaseUrl}'>CritiqueBrainz</a>  profile.`}
+    >
+      {" "}
+      <FontAwesomeIcon icon={faInfoCircle as IconProp} />
+    </button>
+  );
+
+  constructor(props: CBReviewModalProps) {
+    super(props);
+    this.state = {
+      entityToReview: null,
+      loading: false,
+      reviewValidateAlert: null,
+
+      releaseGroupEntity: null,
+      artistEntity: null,
+      recordingEntity: null,
+
+      textContent: "",
+      rating: 0,
+      language: "en",
```

Comment:
An open question to everybody: do we want to default to English?
There might be ways to [detect the user's preferred language](https://caniuse.com/mdn-api_navigatorlanguage_languages) from the navigator, and we could always fall back to english for non-supported browsers.

Reply:
I think it's a good idea, it will save non english users an extra click every time they want to review. 
I will get this implemented in a separate PR once this one is merged so this one doesn't get too bloated.

Reply:
I wonder if there would be a way to use the user's browser language, and defaulting to english if it's not accessible.
Perhaps something with `const locale = navigator.language || navigator.userLanguage || "en";`, although I haven't checked how well the language codes match.
For example for my browser I get `fr-FR`; is that accepted, or should it be `fr` only?

Anyway, something for another PR !

Reply:
@MonkeyDo Yeah you're right, it should be the two letter iso-639-1 code `fr` only. There's no direct way to get it from the browser it looks like but we can use something like `navigator.language.substring(0,2)`.

Reply:
Or maybe split on characters `-` (and maybe `_` I'd need to review language codes), since I assume some languages will have more than 2 characters